tests/testthat/test-best_case_worst_case_emission_profile.R

```diff
@@ -0,0 +1,86 @@
+test_that("Three `ep_products` with the same `benchmark` but with different `emission_profile` will have `best_case` only for the low `emission_profile` product and have `worst_case` only for the high `emission_profile` product", {
+  example_data <- best_case_worst_case_emission_profile_sample()
```

Comment:
This deduplicates some code but seems to be still quite rigid. Here's a more flexible alternative that you might want to consider.

<details/>
<summary/>
reprex
</summary>

``` r
  devtools::load_all()
#> ℹ Loading tiltIndicatorAfter
  
  # It's too rigid
  try(best_case_worst_case_emission_profile_sample(
    ep_product = NA_character_,
    emission_profile = NA_character_
  ))
#> Error in best_case_worst_case_emission_profile_sample(ep_product = NA_character_,  : 
#>   unused arguments (ep_product = NA, emission_profile = NA)
  
  # Really flexibble
  best_case_worst_case_emission_profile_sample_impl <- function(companies_id = "any",
                                                                ep_product = c("one", "two", "three"),
                                                                benchmark = "all",
                                                                emission_profile = c("low", "medium", "high")) {
    tibble::tibble(
      companies_id = companies_id,
      ep_product = ep_product,
      benchmark = benchmark,
      emission_profile = emission_profile
    )
  }
  best_case_worst_case_emission_profile_sample <- tiltIndicator:::example_data_factory(best_case_worst_case_emission_profile_sample_impl())
  
  
  best_case_worst_case_emission_profile_sample()
#> # A tibble: 3 × 4
#>   companies_id ep_product benchmark emission_profile
#>   <chr>        <chr>      <chr>     <chr>           
#> 1 any          one        all       low             
#> 2 any          two        all       medium          
#> 3 any          three      all       high
  
  # Modify
  best_case_worst_case_emission_profile_sample(ep_product = NA_character_)
#> # A tibble: 3 × 4
#>   ep_product companies_id benchmark emission_profile
#>   <chr>      <chr>        <chr>     <chr>           
#> 1 <NA>       any          all       low             
#> 2 <NA>       any          all       medium          
#> 3 <NA>       any          all       high
  
  # Remove
  best_case_worst_case_emission_profile_sample(ep_product = NULL)
#> # A tibble: 3 × 4
#>   companies_id ep_product benchmark emission_profile
#>   <chr>        <chr>      <chr>     <chr>           
#> 1 any          one        all       low             
#> 2 any          two        all       medium          
#> 3 any          three      all       high
  
  # Add
  best_case_worst_case_emission_profile_sample(new = 1)
#> # A tibble: 3 × 5
#>     new companies_id ep_product benchmark emission_profile
#>   <dbl> <chr>        <chr>      <chr>     <chr>           
#> 1     1 any          one        all       low             
#> 2     1 any          two        all       medium          
#> 3     1 any          three      all       high
  
  # Use indirect, stable references to less-stable column names
  # R/column-names.R
  europages_product <- function() "ep_product"
  
  # tests/testthat/test-best_case_worst_case.R
  best_case_worst_case_emission_profile_sample(!!europages_product() := NA_character_)
#> # A tibble: 3 × 4
#>   ep_product companies_id benchmark emission_profile
#>   <chr>      <chr>        <chr>     <chr>           
#> 1 <NA>       any          all       low             
#> 2 <NA>       any          all       medium          
#> 3 <NA>       any          all       high
```

</details>

Reply:
Implemented the change in this [commit](https://github.com/2DegreesInvesting/tiltIndicatorAfter/pull/243/commits/ec0bdb4e13c64560cea94306647da8e9b5a4762a)